From dba74289f74095944b39fc96b069c512b5321b7a Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sun, 12 May 2024 14:53:45 +0200
Subject: [PATCH] Use BUILD_BUG_ON instead of checking size at runtime for
 pcie_tx_init_ndp

Use BUILD_BUG_ON instead of checking size at runtime for
pcie_tx_init_ndp.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 hif/pcie/8964/tx_ndp.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

--- a/hif/pcie/8964/tx_ndp.c
+++ b/hif/pcie/8964/tx_ndp.c
@@ -334,13 +334,8 @@ int pcie_tx_init_ndp(struct ieee80211_hw
 	struct ieee80211_tx_info *tx_info = IEEE80211_SKB_CB(&skb);
 	int rc;
 
-	if (sizeof(struct pcie_tx_ctrl_ndp) >
-	    sizeof(tx_info->driver_data)) {
-		wiphy_err(hw->wiphy, "driver data is not enough: %zu (%zu)\n",
-			  sizeof(struct pcie_tx_ctrl_ndp),
-			  sizeof(tx_info->driver_data));
-		return -ENOMEM;
-	}
+	BUILD_BUG_ON(sizeof(struct pcie_tx_ctrl_ndp) >
+		     sizeof(tx_info->driver_data));
 
 	rc = pcie_tx_ring_alloc_ndp(priv);
 	if (rc) {
